pkg/autoupdate/findnew.go

```diff
@@ -266,6 +263,41 @@ func getUpdateDir(fullBinaryPath string) string {
 	return fmt.Sprintf("%s%s", components[0], updateDirSuffix)
 }
 
+// Find the possible updates. filepath.Glob returns a list of things
+// that match the requested pattern. We sort the list to ensure that
+// we can tell which ones are earlier or later (remember, these are
+// timestamps).
+func getPossibleUpdates(updateDir, binaryName string) ([]string, error) {
+	// If this is launcher running on macOS, then we should have app bundles available instead --
+	// check for those first.
+	if runtime.GOOS == "darwin" {
+		binarySuffix := filepath.Join("Contents", "MacOS", binaryName)
+		fileGlob := filepath.Join(updateDir, "*", "*.app", binarySuffix)
+		possibleUpdates, err := filepath.Glob(fileGlob)
+
+		if err == nil && len(possibleUpdates) > 0 {
```

Comment:
I kinda get it -- this is a narrow carveout that's returning the happy path for macOS app bundles. But it feels a bit weird. It inverts the usual early-return-on-error logic. I'm not sure if there's something simpler. 

Also maybe worth noting that `err` on glob indicates some massive system failure.

Reply:
James mentioned re the globbing error [here](https://github.com/kolide/launcher/pull/932#discussion_r1021942621) -- I will update to add some logging so that it's clear what the issue is in that case. I could add more documentation here as well if that would make it clearer.